### ISSUE TITLE
Using MathJax version 3 for documentation

### ISSUE THREAD

**Issue by albert-github** [CGAL/cgal#5707]
Currently the documentation uses MathJax version 2 to render formulas. Since some time MathJax version 3 exists and in due time the MathJax version 2 will not be available anymore ("not likely to be until 2022", see: https://groups.google.com/g/mathjax-users/c/MKQ9TdVXSSo). Since a little while the doxygen master supports MathJax version3 (see https://github.com/doxygen/doxygen/issues/7346 and https://github.com/doxygen/doxygen/pull/8496).

The construct used to see which version of MathJax to be used (from CDN or local) is quite complex, so no pull request but some code that indicates the required changes:
**BaseDoxyfile**
*version 2*
```
USE_MATHJAX            = YES
MATHJAX_FORMAT         = HTML-CSS
MATHJAX_RELPATH        = https://cdn.jsdelivr.net/npm/mathjax@2/
MATHJAX_EXTENSIONS     = TeX/AMSmath TeX/AMSsymbols
MATHJAX_CODEFILE       = CGAL_mathjax.js
```
*version 3*
```
USE_MATHJAX            = YES
MATHJAX_VERSION        = MathJax_3
MATHJAX_FORMAT         = chtml
MATHJAX_RELPATH        = https://cdn.jsdelivr.net/npm/mathjax@3
MATHJAX_EXTENSIONS     = ams
MATHJAX_CODEFILE       = CGAL_mathjax_v3.js
```
where the line regarding `MATHJAX_RELPATH` the line is mentioned above about CDN / local.

Furthermore the `MATHJAX_CODEFILE` syntax has slightly changed so the new file should have the contents as in this attachment: [CGAL_mathjax_v3.js.gz](https://github.com/CGAL/cgal/files/6466483/CGAL_mathjax_v3.js.gz)

**Comment by albert-github** [comment]
After a little discussion in the MathJax discussion forum (see https://groups.google.com/g/mathjax-users/c/oS0yQLb5BMk) a small change was necessary for the attachment in the original issue, it now is: [CGAL_mathjax_v3.js.tar.gz](https://github.com/CGAL/cgal/files/6487818/CGAL_mathjax_v3.js.tar.gz). The fix also requires that the proposed doxygen pull request https://github.com/doxygen/doxygen/pull/8554 is integrated.

**Comment by albert-github** [comment]
The proposed pull request #5715 introduces 2 new macros (that were previously omitted), hence they have also to be added to the v3 version: [CGAL_mathjax_v3_2.js.tar.gz](https://github.com/CGAL/cgal/files/6488953/CGAL_mathjax_v3_2.js.tar.gz)

**Comment by albert-github** [comment]
@MaelRL This issue does not have a pull request, the mentioned pull request is for something different (but those changes have to be introduced in this issue as well, hence the second attachment). There is no pull request like I mentioned in the issue:
> The construct used to see which version of MathJax to be used (from CDN or local) is quite complex, so no pull request but some code that indicates the required changes.

**Comment by sloriot** [comment]
@albert-github do you know if this is still an issue as of today or all is know fine and nothing needs to be done on our side for our upcoming release?

**Comment by albert-github** [comment]
As far as I can tell with the current code base (e90842fd05bbbec3148a45a2a8608f494c2ff075 ) there are no changes in respect to switching to MathJax V3 and there is also no PR yet for this.
For the upcoming release I wouldn't do anything anymore regarding the MathJax version, though for a future release this can / should be an issue to be fixed.

I could create a PR for this but I think this should be done independent of the 6.0 release (I will have to dive into it again).

Note the label "Has Pull Request" should be removed like already indicated in https://github.com/CGAL/cgal/issues/5707#issuecomment-850211266

**Comment by albert-github** [comment]
MathJax version 4 is available and is supported since doxygen 1.15.0 so might be worthwhile to take a broader look at the version of MathJax and doxygen to be used.

(Note MathJax 3 is available in doxygen since version 1.9.2, released 18-08-2021. CGAL uses currently doxygen 1.9.6).

**Comment by albert-github** [comment]
Seen the discussion with https://github.com/CGAL/cgal/pull/9118#discussion_r2742261065, but actually because MathJax 4 is available (https://github.com/CGAL/cgal/issues/5707#issuecomment-3753761605)  it is better to skip version MathJax 3 and directly go to MathJax 4.
Changing title of issue for this.

**Comment by albert-github** [comment]
A description what has to be done to update to MathJax 4, not a PR as there are steps that lie outside my possibilities).

Currently CGAL uses for the documentation MathJax version 2 (actually a mix: downloaded 2.2.1 version for releases,
downloaded 2.7.4 version for the overnight documentation builds and 2.7.9 from the internet for the builds directly
from pull requests in GitHub).
To update CGAL in respect to the documentation generation (first for the master version, but later on also for release
and PR builds) to MathJax 4 a number of actions are required.

- Update settings file (Documentation/doc/resources/1.14.0/BaseDoxyfile.in)
  The file should actually be renamed to `Documentation/doc/resources/1.15.0/BaseDoxyfile.in` as MathJax 4 is available
  for doxygen 1.15.0, so even 1.16.1 would be better).
  ```
  USE_MATHJAX            = YES

  MATHJAX_VERSION = MathJax_v4

  ${CGAL_DOC_MATHJAX_LOCATION_FULL_OPTION_LINE}

  MATHJAX_EXTENSIONS     =

  MATHJAX_CODEFILE       = ${CGAL_DOC_RESOURCE_DIR}/CGAL_mathjax.js
  ```
- Replace `CGAL_mathjax.js` with ([CGAL_mathjax.js](https://github.com/user-attachments/files/25049495/CGAL_mathjax.js)):
  ```
  window.MathJax = {
  tex:
    {
    tags: 'ams',
    macros: {
          qprel: [ "{\\gtreqless}", 0],
          qpx: [ "{\\mathbf{x}}", 0],
          qpl: [ "{\\mathbf{l}}", 0],
          qpu: [ "{\\mathbf{u}}", 0],
          qpc: [ "{\\mathbf{c}}", 0],
          qpb: [ "{\\mathbf{b}}", 0],
          qpy: [ "{\\mathbf{y}}", 0],
          qpw: [ "{\\mathbf{w}}", 0],
          qplambda: [ "{\\mathbf{\\lambda}}", 0],
          ssWpoint: [ "{\\bf #1}", 1],
          ssWeight: [ "{w_{#1}}", 1],
          dabs: [ "{\\parallel\\! #1 \\!\\parallel}", 1],
          E: [ "{\\mathrm{E}}", 0],
          A: [ "{\\mathrm{A}}", 0],
          R: [ "{\\mathrm{R}}", 0],
          N: [ "{\\mathrm{N}}", 0],
          Q: [ "{\\mathrm{Q}}", 0],
          Z: [ "{\\mathrm{Z}}", 0],
          ccSum: [ "{\\sum_{#1}^{#2}{#3}}", 3],
          ccProd: [ "{\\prod_{#1}^{#2}{#3}}", 3],
          pyr: [ "{\\operatorname{Pyr}}", 0],
          aff: [ "{\\operatorname{aff}}", 0],
          Ac: [ "{\\cal A}", 0],
          Sc: [ "{\\cal S}", 0],
        }
    }
  };
  ```
- Adjust in `Documentation/doc/scripts/test_doxygen_versions.sh` the lines with:
  ```
  ... -DCGAL_DOC_MATHJAX_LOCATION:STRING=../../MathJax
  ```
  to
  ```
  ... -DCGAL_DOC_MATHJAX_LOCATION:STRING=../../MathJax_v4
  ```
  ... -DCGAL_DOC_MATHJAX_LOCATION:STRING=../../MathJax
  ```
- Download the MathJax 4 release (preferably 4.4.1 or newer) and place this as `MathJax_v4` parallel to `MathJax`
  mentioned above
- Adjust the `\build` command for PR documentation builds to reflect the switch to MathJax 4
- Check `Documentation/doc/CMakeLists.txt` as here we see:
  ```
  # set up the directories and variables
  if(CGAL_DOC_MATHJAX_LOCATION)
    set(CGAL_DOC_MATHJAX_LOCATION_FULL_OPTION_LINE
        "MATHJAX_RELPATH = ${CGAL_DOC_MATHJAX_LOCATION}")
  else()
    set(CGAL_DOC_MATHJAX_LOCATION
        "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2")
  endif()
  ```
  not sure what has to be done here
